.circleci/config.yml

```diff
@@ -0,0 +1,103 @@
+version: 2
+jobs:
+  build-flower:
+    working_directory: ~/flower
+    machine:
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - run:
+          name: Create cache directory
+          command: mkdir -p docker/images/flower
+
+      - restore_cache:
+          keys:
+            - flower-image-{{ .Revision }}
+            - flower-image
+
+      - run:
+          name: build flower image
+          command: docker build -t fundocker/flower:${CIRCLE_SHA1} .
+
+      - run:
+          name: List available images
+          command: docker images fundocker/flower
+
+      - run:
+          name: Store docker image in cache
+          command: |
+            docker save \
+              -o docker/images/flower/flowr.tar \
+              fundocker/flower:${CIRCLE_SHA1}
+
+      - save_cache:
+          paths:
+            - ~/flower/docker/images/flower
+          key: flower-image-{{ .Revision }}
+
+  hub:
+    working_directory: ~/flower
+    machine: true
+    steps:
+      - checkout
+      - run:
+          name: Create cache directory
+          command: mkdir -p docker/images/flower
+
+      - restore_cache:
+          keys:
+            - flower-image-{{ .Revision }}
+
+      - run:
+          name: Load images to docker engine
+          command: |
+            docker load < docker/images/flower/flower.tar
```

Comment:
From my current experience with small builds as this one, rebuilding the image in the publication job is more efficient that saving/loading an archived image. I would remove those steps.